pkg/isoeditor/stream.go

```diff
@@ -29,7 +47,69 @@ func NewRHCOSStreamReader(isoPath string, ignitionContent *IgnitionContent, ramd
 		return nil, err
 	}
 
-	r, err := readerForFileContent(isoPath, ignitionImagePath, isoReader, ignitionReader)
+	// Starting with version 0.17.0 of the CoreOS installer a new '/coreos/igninfo.json' file may
+	// exist to indicate the location of the ignition inside the ISO. For example, in the S390
+	// platform it will contain something like this:
+	//
+	//	{
+	//		"file": "images/cdboot.img",
+	//		"offset": 66497660,
+	//		"length": 262144
+	//	}
+	//
+	// If it exists we need to read it, otherwise we can use the '/images/ignition.img' location
+	// used in previous versions.
+	var ignInfo IgnInfo
+	ignInfoBytes, err := ReadFileFromISO(isoPath, ignitionInfoPath)
```

Comment:
Can't we validate exact error or we always want to retry?

Reply:
Unfortunatelly the diskfs library doesn't return `os.ErrNotExist`, it returns a generic string error with this message if the file doesn't exist:

```
could not read directory entries for /coreos
```

I could check that string, but I thought it is fragile and not worth. Trying always with the default doesn't hurt.